src/main/java/Main.java

```diff
@@ -10,58 +12,128 @@
 import goblintserver.GoblintServer;
 
 import org.eclipse.lsp4j.jsonrpc.messages.Either;
-
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class Main {
 
+    private static final String gobPieConfFileName = "gobpie.json";
     private static final Logger log = LogManager.getLogger(Main.class);
+    private static MagpieServer magpieServer;
 
     public static void main(String... args) {
 
-        // launch the server only if there is a GobPie conf file present
-        if (new File(System.getProperty("user.dir") + "/" + "gobpie.json").exists()) {
-
-            MagpieServer magpieServer = createMagpieServer();
-
-            if (magpieServer == null) {
-                log.info("Unable to launch MagpieBridge.");
-            } else {
-                magpieServer.launchOnStdio();
-                log.info("MagpieBridge server launched.");
-                magpieServer.doAnalysis("c", true);
+        int exit = createMagpieServer();
+        launchMagpieServer();
+
+        if (exit != 0) {
+            magpieServer.forwardMessageToClient(
+                new MessageParams(MessageType.Error, "GobPie extension is unable to analyse the code. Please check the output terminal of GobPie extension for more information."));
+            switch (exit) {
+                case 1:
+                    log.error("Unable to fully start GobPie extension: Starting Goblint Server failed.");
```

Comment:
breaks, please

Reply:
Besides that, an enum would be better than magic numbers that have to match between two different classes.

Or possibly even just a custom exception type that gets caught at the top level and turned into a UI message (and logged). Then the error messages don't have to be maintained separately from where the errors actually happen.